docs/developer-docs/smart-contracts/signatures/signing-messages-tschnorr.mdx

```diff
@@ -0,0 +1,308 @@
+
+## Signing messages
+
+Canisters can sign threshold Schnorr messages without holding the Schnorr keys themselves. Each key is derived from a master key held by the dedicated threshold Schnorr subnet. When a canister needs a signature, a request is sent to the threshold Schnorr subnet that computes the signature based on the requesting canister's root key and then returns the signature to the canister.
+
+To sign a message, the `sign_with_schnorr` method is used, which requires that you specify the Schnorr algorithm to be used, either `bip340secp256k1` or `ed25519`.
+
+To deploy code using the `sign_with_schnorr` method, you will need to specify the correct `key_ID` for the environment in the canister's source code. The following key IDs are supported:
+
+- `dfx_test_key1`: Only available on the local replica started by dfx.
+- `test_key_1`: Test key available on the ICP mainnet.
+- `key_1`: Production key available on the ICP mainnet.
+
+<AdornedTabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+  public shared ({ caller }) func sign(message_arg : Text, algorithm_arg : SchnorrAlgotirhm) : async {
+    #Ok : { signature_hex : Text };
+    #Err : Text;
+  } {
+    try {
+      Cycles.add(25_000_000_000);
+      let { signature } = await ic.sign_with_schnorr({
+        message = Text.encodeUtf8(message_arg);
+        derivation_path = [Principal.toBlob(caller)];
+        key_id = { algorithm = algorithm_arg; name = "dfx_test_key" };
+      });
+      #Ok({ signature_hex = Hex.encode(Blob.toArray(signature)) });
+    } catch (err) {
+      #Err(Error.message(err));
+    };
+  };
+```
+
+</TabItem>
+
+<TabItem value="rust" label="rust">
+
+```rust
+#[update]
+async fn sign(message: String, algorithm: SchnorrAlgorithm) -> Result<SignatureReply, String> {
+    let internal_request = ManagementCanisterSignatureRequest {
+        message: message.as_bytes().to_vec(),
+        derivation_path: vec![ic_cdk::api::caller().as_slice().to_vec()],
+        key_id: SchnorrKeyIds::TestKeyLocalDevelopment.to_key_id(algorithm),
+    };
+
+    let (internal_reply,): (ManagementCanisterSignatureReply,) =
+        ic_cdk::api::call::call_with_payment(
+            mgmt_canister_id(),
+            "sign_with_schnorr",
+            (internal_request,),
+            25_000_000_000,
+        )
+        .await
+        .map_err(|e| format!("sign_with_schnorr failed {e:?}"))?;
+
+    Ok(SignatureReply {
+        signature_hex: hex::encode(&internal_reply.signature),
+    })
+}
+```
+
+</TabItem>
+</AdornedTabs>
+
+### Cost
+
+The `sign_with_schnorr` method requires cycles to be attached. When the method in the Rust CDK is used, the required cycles are automatically attached. In Motoko, the cycles need to be added manually.
```

Comment:
CDK doesn't support Schnorr yet, so we are constructing the call manually. But this will be supported at some point and the statement will be valid in the future.

Reply:
Removed this statement for now